### PR TITLE
Actuators

### PR DESCRIPTION
**Pull request by coletaylor788**
Adding controller node that outputs a CarPose and a CarState
Added files from buzzmobile-old for the arduino
Added CarState message

### CODE REVIEW

**Review by dacohen — APPROVED** [review]

File: buzzmobile/drive/horn/src/horn_node.cpp
Comment (irapha) [inline buzzmobile/drive/horn/src/horn_node.cpp]: This doesn't seem to compile:
https://travis-ci.org/gtagency/buzzmobile/builds/168462865

Maybe we need to `<depend>` on the manifest?

File: README.md
Comment (irapha) [inline README.md:62]: Can you add instructions for how the controller changes CarPose? Probs in README (in a new section, called `Manual Driving`, just like we have `Environment` and `Running`). Also mention how to change manual to auto modes

File: README.md
Comment (irapha) [inline README.md]: Can you add this `ros-indigo-joy` dependency to the `buzzmobile/package.xml`? Then I don't think we need to have it here because we can just install all deps declared there. All other deps like lidar and gps are there already.

Also, if `ds4drv` is a thing that's available as a rosdep (we'd have to look it up), then put it there too. Then we don't need this addition to the `README`.

**Review by irapha — CHANGES_REQUESTED** [review]

**Comment by coletaylor788** [comment]
Whenever you create a ros msg that gets used by a cpp node, you need to add dependency to the CMakeLists.txt file for each Node that uses a custom message so it waits until after the messages are generated to compile. Otherwise catkin_make tries to compile the node .cpp file before the custom message .h file has been created. catkin_make aggresively attempts to parallelize.

add_dependencies(your_node ${catkin_EXPORTED_TARGETS})

http://answers.ros.org/question/62092/catkin_make-started-failing-to-generate-h-from-msg/

**Comment by coletaylor788** [comment]
Added joy as a ros dependency in package.xml and updated .travis.yaml to install the ros dependencies. Note: I needed to source the script in catkin_ws/devel before installing the ros dependencies so it would recognize buzzmobile. Since I don't think devel exists until after catkin_make, I put this after the make command. (Not sure if there is a better alternative)

File: .travis.yml
Comment (irapha) [inline .travis.yml:35]: should we be installing dependencies before making? I'd assume we install deps in `before_script`, no?

**Review by irapha — CHANGES_REQUESTED** [review]

File: .travis.yml
Comment (irapha) [reply .travis.yml:35]: As discussed offline, you can only install deps after you source the `setup.bash`, which is in the `devel/`, which is generated from `catkin_make`. So the way this is done seems correct.

At one point, when we have rostests ( see #20 ) we will move these three script commands to `before_script` and have the test command be the `script`.

File: buzzmobile/sense/controller/src/controller_node.cpp
Comment (joshuamorton) [inline buzzmobile/sense/controller/src/controller_node.cpp]: Factor this out into constants.yml?

File: buzzmobile/sense/controller/src/controller_node.cpp
Comment (joshuamorton) [inline buzzmobile/sense/controller/src/controller_node.cpp]: I think we actually shouldn't need that, ROS will cache old values for us, but not a major issue.

**Review by joshuamorton — CHANGES_REQUESTED** [review]

File: buzzmobile/sense/controller/src/controller_node.cpp
Comment (irapha) [reply buzzmobile/sense/controller/src/controller_node.cpp]: Hm. Idk what happens if we go with either case. If we don't continuously publish a message to the motor, won't the motor get the message and then stop doing what it was doing because it didn't get a message right after telling it to continue at it?

If this doesn't happen, then I agree this thing isn't needed.

File: buzzmobile/sense/controller/src/controller_node.cpp
Comment (irapha) [reply buzzmobile/sense/controller/src/controller_node.cpp]: See discussion on #16

**Review by joshuamorton — APPROVED** [review]

**Comment by irapha** [comment]
Can you merge master into actuators and fix conflicts?

File: README.md
Comment (irapha) [inline README.md]: Mention something about car starting in START mode, and that you need to switch to either MANUAL or AUTO mode in the controller?

File: README.md
Comment (irapha) [inline README.md]: Change title to "Manual Mode Controls"?

File: README.md
Comment (irapha) [inline README.md]: controller_node => `controller_node`

File: buzzmobile/CMakeLists.txt
Comment (irapha) [inline buzzmobile/CMakeLists.txt]: Didn't we remove `horn` node?

File: buzzmobile/drive/arduino/src/car_driver/car_driver.ino
Comment (irapha) [inline buzzmobile/drive/arduino/src/car_driver/car_driver.ino]: Don't merge to master with this commented out. We'll wait on @sastorer to fix the serial 6 connection and test this before we merge this PR.

File: buzzmobile/drive/car_interface/CMakeLists.txt
Comment (irapha) [inline buzzmobile/drive/car_interface/CMakeLists.txt]: Do we need all this commented out stuff?

File: buzzmobile/drive/car_interface/src/main.cpp
Comment (irapha) [inline buzzmobile/drive/car_interface/src/main.cpp:73]: Just a note that I'm okay with the commented out stuff on this file since there's an open issue for adding this in the future #41 #66

File: buzzmobile/drive/horn/CMakeLists.txt
Comment (irapha) [inline buzzmobile/drive/horn/CMakeLists.txt]: Do we need this commented out stuff?

File: buzzmobile/drive/horn/src/horn_node.cpp
Comment (irapha) [inline buzzmobile/drive/horn/src/horn_node.cpp]: Also fine with these commented out things because we have open issues for handling these in the future #41 #66

File: buzzmobile/plan/car_pose_mux/car_pose_mux.py
Comment (irapha) [inline buzzmobile/plan/car_pose_mux/car_pose_mux.py]: This will only print either MANUAL or AUTO. It should also print START.

File: buzzmobile/plan/controller/CMakeLists.txt
Comment (irapha) [inline buzzmobile/plan/controller/CMakeLists.txt]: so many comments y pls

File: buzzmobile/plan/controller/src/controller_node.cpp
Comment (irapha) [inline buzzmobile/plan/controller/src/controller_node.cpp:173]: We might want another set of eyes in this file. @joshuamorton @dacohen Do you guys know what this commented out stuff does? They were in the controller in `buzzmobile-old`.

**Review by irapha — CHANGES_REQUESTED** [review]

**Comment by joshuamorton** [comment]
One of the nice things Ros does is that messages are stored in a queue, and
that queue will never be empty once it is filled, that's why ros.spinonce
works.

File: buzzmobile/plan/car_pose_mux/car_pose_mux.py
Comment (irapha) [inline buzzmobile/plan/car_pose_mux/car_pose_mux.py:37]: Maybe factor out this logging? This method got kinda unnecessarily big

**Review by irapha — CHANGES_REQUESTED** [review]

**Review by irapha — APPROVED** [review]

**Pull request merged**
